src/Sarif/EnumeratedArtifact.cs

```diff
@@ -62,9 +62,14 @@
             {
                 if (sizeInBytes != null) { return sizeInBytes.Value; };
 
-                this.sizeInBytes = Uri!.IsFile
-                    ? (ulong)FileSystem.FileInfoLength(Uri.LocalPath)
-                    : (ulong?)null;
+                if (!File.GetAttributes(Uri.GetFilePath()).HasFlag(FileAttributes.Directory))
```

Comment:
>!File.GetAttributes(Uri.GetFilePath()).HasFlag(FileAttributes.Directory) [](http://example.com/codeflow?start=20&length=72)

thanks for explaining the reason in the description.
The fix itself doesn't look straight forward, I will need to get the branch and try it.
I think, at least if the result of the Uri.GetFilePath() not a valid file path then File.GetAttributes should throw unhandled exception here.

Reply:
Right none of this code is hardened for scenarios such as in-memory analysis, scanning from zip archives, etc. These won't be file scenarios.

But I'm not sure how to handle it.

Suvam, I see that I was very off-base in our conversation, the root issue appears to be someone attempting to retrieve a file size from a directory, not an attempt to retrieve file size from a relative path causing an exception.